dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/Servers.scala

```diff
@@ -156,7 +165,7 @@ private[lagom] object Servers {
         val serverClass = loader.loadClass("com.lightbend.lagom.internal.cassandra.CassandraLauncher")
         server = serverClass.newInstance().asInstanceOf[Server]
 
-        server.start(directory, "dev-embedded-cassandra.yaml", cleanOnStart, port, jvmOptions.toArray)
+        server.start(directory, yamlConfig.orNull, cleanOnStart, port, jvmOptions.toArray)
```

Comment:
This is very awkward. I wasn't able to keep it as `Option[File]` because of the usage of structural type `type Server`. Somehow I get a runtime error when calling the start method of `CassandraLauncher`. 

Lost a stupid amount of time trying to figure out the issue and then decided to workaround. In general I believe we should move out of structural type. This is only make the code obscure. As far I could understand, the only reason for this is to use be able to resume some minimal code, eg: `tryStop`. This is not worth the complexity and the hurdle. 

I will remove it on another PR.

Reply:
I agree in principle, but I'm not so sure it will be so straightforward to remove. These class loaders are isolated from each other and I'm not sure where to define a type that will be available to both. I think this relies on the fact that method calls on structural types are implemented via reflection.

Reply:
Do you remember what the runtime error was?

Reply:
The error was a `NoSuchMethodException` and I guess it was related to the usage of generics, which actually doesn't make much sense for me. 

I'm not sure if structural typing is being used because of classloader issues. I didn't think about it. If that's the case, we should really add a comment explaining it. That's totally not obvious at first sight, at least not for me.

Reply:
I bet it's because the Scala `Option` class is also loaded by different classloaders. In fact, they might not even be binary compatible (2.10 vs 2.11).

I wonder if it would work if you convert to Java `Optional`... maybe that's even worse than just using `null`!

Reply:
I did more test on this. And I can confirm it's a classloader issue. Also tried to use `Optional`, but it's also awkward. The problem I see is that the knowledge is spread and it's not obvious why we are using `null` or `Optional`. 

It's also not clear why we are fabricating another classloader for it. It wasn't the case in `1.3.10`. The `ServiceLocator` where started with another classloader, but Kafka and Cassandra were using the current classloader. In `1.4.0`, @jroper changed it, but only for Cassandra. 

@jroper could you shed some light on this? 

I see that we are using `ScalaClassLoaderManager` to build a Scala classloader cache and that it's being shared by the `ServiceManager` and the `ServerMojos` in the maven plugin, but why? Why aren't we using the current classloader? I tested it and it simply works. Am I missing something here?

Reply:
I'm surprised that it works... how did you test? This file `Servers.scala` is in the `build-tool-support` module, which is built only for Scala 2.10. `CassandraLauncher` in the `cassandra-server` module cross-builds for 2.11 and 2.12. I think that's why they use isolated classloaders. They also need very different class paths.

In 1.3.10 this worked completely differently, forking off a separate process instead. The change is explained by the commit comment in https://github.com/lagom/lagom/pull/808/commits/769886974edf5ea819c6da7dec59a6ecee757b8d

It probably is worth adding comments to the code as well.

Reply:
I tested it by simply building locally and using it in a hello world app. 

In 1.3.10, we start a new process pretty much like calling `java -cp ...` . We build the value for `cp`  using a `Seq[File]` that we get from sbt. This list contains all the jars needed for a specific server. The process runs using a brand new classloader.

In 1.4.x, we do it a little differently. We build a new classloader using the same `Seq[File]` and the `parentClassloader` which is actually the classloader used by the compiler. 

The problem appears because of the structural type `Server` and its `start()` method.  If we use a non-std lib class as argument the reflection call won't work because `scala.Option` on this classloader is "not the same" as the one in the `parentClassloader`. 

Later on, inside the `CassandraLauncher`, we will build another classloader the same way we were doing in `1.3.10` and spawn a new process. So, the problem is not the classloader used to run the server, but the one we fabricate to reach `CassandraLauncher` from inside `build-tool-support`.

Basically, what we are doing here is creating a bridge from `build-tool-support` to `cassandra-server` in order to access `CassandraLauncher`.  My initial question is why we did it by using the `ScalaClassLoaderManager` and compiler classloader. Why we need them to build that bridge?

Reply:
I will push the code I have up to now. If needed, we can revert that single commit.